dotcom-rendering/src/model/extractTrendingTopics.ts

```diff
@@ -0,0 +1,74 @@
+import type { FECollectionType, FEFrontCard } from '../types/front';
+import type { FETagType } from '../types/tag';
+
+const allTrails = (collections: FECollectionType[]): FEFrontCard[] =>
+	collections.flatMap((collection) => [
+		...collection.curated,
+		...collection.backfill,
+	]);
+
+const allTrailsDeduped = (allTrailsInput: FEFrontCard[]): FEFrontCard[] =>
```

Comment:
I really like breaking code into small functions that do one thing - especially in cases like this when there's a lot going on! As a general suggestion, there are a couple of things that I would do differently to improve readability:

1. I would name some of these functions differently to make them a bit more generic and independent from the whole calculation.
2. I would merge some of these functions. 
3. Add unit tests to demonstrate the intention of the code.

For example (first is how it currently is - second is suggestion):

```
const allTrailsDeduped = (allTrailsInput: FEFrontCard[]): FEFrontCard[]
const dedupeTrails = (trails: FEFrontCard[]): FEFrontCard[]

const allTags = (allTrailsDedupedInput: FEFrontCard[]): FETagType[]
const getTags = (trails: FEFrontCard[]): FETagType[]

const desiredTags = (allTagsInput: FETagType[]): FETagType[] 
const getKeywordTags = (tags: FETagType[]): FETagType[] => 
     tags.filter( tag => 
                       tag.properties.paidContentType?.includes('Keyword') ??
	                  tag.properties.paidContentType?.includes('Topic') ??
	                  tag.properties.tagType === 'Keyword');
```